### PR TITLE
Add basic debug config

### PR DESCRIPTION
**Pull request by apeltzer**
This adds a (very basic tbh atm) debug config profile. We can use this to e.g. test where individual processes were submitted to, and can try to extend it to include more things that I didn't include yet. 

Not sure, whether we can also change `NXF_DEBUG` levels with something like that - I suppose unfortunately not. @pditommaso might know?

### CODE REVIEW

**Comment by pditommaso** [comment]
Nextflow trace logging can only enabled via `-trace` command line option. `NXF_DEBUG` enables the debugging of task wrapper, however rarely it's useful and in any case cannot be define by the config file either. 

For application debugging can be useful the [dump] (https://www.nextflow.io/docs/latest/operator.html#dump) operator that can be used to debug operator. This as a replacement for [this](https://github.com/SciLifeLab/Sarek/blob/8a2a181640f152192c53ebe8733708100c5faa62/main.nf#L112-L116) snippet.

**Comment by MaxUlysse** [comment]
@pditommaso Thanks a lot for this `dump` operator, I'll be sure to update Sarek.
But I kinda like my snippets, it has a really nice display ;-)

**Comment by pditommaso** [comment]
In Italian there's an idiomatic phrase that sounds like: every cockroach is
nice for his mom .. 😂😂

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/nextflow.config [hunk 1]
```diff
@@ -47,6 +47,9 @@ profiles {
     includeConfig 'conf/base.config'
     includeConfig 'conf/test.config'
   }
+  debug {
+    includeConfig 'conf/debug.config'
```
[the remark this answers is not in the document]
Comment (apeltzer) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/nextflow.config]: ATM its just one line yes, but I thought we could maybe make other things possible in NXF soon (e.g. specifying the trace/debug modes that Paolo mentioned)... that would be the coolest thing to do IMHO.

[the remark this answers is not in the document]
Comment (apeltzer) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/nextflow.config]: Can change it to a single line though for now...?

[the remark this answers is not in the document]
Comment (MaxUlysse) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/nextflow.config]: It's your PR, you decide, we argue, you change, we argue again... ;-)

[the remark this answers is not in the document]
Comment (apeltzer) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/nextflow.config]: I've removed it - we can add that once there is more stuff for debugging settable via a config scope.

**Comment by ewels** [comment]
> Okay, so we can't add a possibility to e.g. have this in our profile to dump the output of channels when this profile is used? Would be cool to just mix-in the debug profile with `-profile debug,<yourotherprofiles>` and just get the debug output, too - also for getting feedback from users if things break?

What about using the Sarek snippet but in the conditional test if the active profile is `debug`? Then no additional command line flags required.

**Comment by apeltzer** [comment]
I like the idea - @MaxUlysse do you think we should take them the same way or use the `dump` operator too as Paolo mentioned it?

**Comment by MaxUlysse** [comment]
Oooh
I do like my cockroaches as @pditommaso describe them.
They do provide a nice output, and you can definitively access them with a `--verbose` param

**Comment by apeltzer** [comment]
Then I'll also like them and favor them for now ;-)

**Comment by MaxUlysse** [comment]
But on the other end, you have to make that for every single channel you want to display

**Comment by ewels** [comment]
Maybe we should replace `-profile debug` with `--verbose`? I'd like to have one command line flag to do this, and as it's always a combinatorial profile which does very little, it may make more sense to have it as a simple `params` boolean...

**Comment by apeltzer** [comment]
@MaxUlysse But adding `.dump(tag: name)` to every channel is also something we'd have to do... 

@ewels Sounds also good to me, just wanted to have something to turn on/off debugging as it turned out to be quite useful for some cases where I had to look around what happened in pipelines...

**Comment by MaxUlysse** [comment]
I would love to have different level of verbosity also

**Comment by ewels** [comment]
`.dump()` only works with `-dump-channels` though, so we end up with a bunch of different cli flags again if we're not careful (`-profile debug --verbose -dump-channels`).

@MaxUlysse when would different levels be helpful? I mostly see this as either debugging a problem or not. So max verbosity or nothing.

**Pull request merged**